cmd/deploy/deploy.go

```diff
@@ -303,61 +324,48 @@ func (dc *DeployCommand) RunDeploy(ctx context.Context, deployOptions *Options)
 			SkipIfExists: !deployOptions.Dependencies,
 		}
 		if err := pipelineCMD.ExecuteDeployPipeline(ctx, pipOpts); err != nil {
+			if errStatus := updateConfigMapStatus(ctx, cfg, c, data, err); err != nil {
+				return errStatus
+			}
+
 			return err
 		}
 	}
 
+	var errBuild error
 	if deployOptions.Build {
 		buildOptions := &types.BuildOptions{
 			EnableStages: true,
 			Manifest:     deployOptions.Manifest,
 			CommandArgs:  deployOptions.servicesToDeploy,
 		}
 		oktetoLog.Debug("force build from manifest definition")
-		if err := dc.Builder.Build(ctx, buildOptions); err != nil {
-			return err
-		}
+		errBuild = dc.Builder.Build(ctx, buildOptions)
 	} else {
-		svcsToBuild, err := dc.Builder.GetServicesToBuild(ctx, deployOptions.Manifest, deployOptions.servicesToDeploy)
-		if err != nil {
-			return err
-		}
-		if len(svcsToBuild) != 0 {
+		svcsToBuild, errBuild := dc.Builder.GetServicesToBuild(ctx, deployOptions.Manifest, deployOptions.servicesToDeploy)
```

Comment:
Using the same error in multiple checks is very prone to bugs. I'd like to see us find a pattern that doesn't use such a largely scoped error.

I believe the else here is handled right away and can be a new var.

Also, I think we can probably make the error handling a function and handle the de-duplication of error handling by calling a function rather than a hoisted error that spans multiple if/else.